server/sonar-server/src/main/java/org/sonar/server/computation/step/FeedMeasureComputers.java

```diff
@@ -216,12 +224,22 @@ public boolean apply(@Nullable String metric) {
 
   private class ValidateOutputMetric implements Predicate<String> {
     @Override
-    public boolean apply(@Nullable String metric) {
-      if (CORE_METRIC_KEYS.contains(metric)) {
-        throw new IllegalStateException(String.format("Metric '%s' cannot be used as an output metric as it's a core metric", metric));
-      }
-      if (!pluginMetricKeys.contains(metric)) {
-        throw new IllegalStateException(String.format("Metric '%s' cannot be used as an output metric as no plugin declare this metric", metric));
+    public boolean apply(@Nonnull String metric) {
+      checkState(!CORE_METRIC_KEYS.contains(metric), String.format("Metric '%s' cannot be used as an output metric as it's a core metric", metric));
+      checkState(pluginMetricKeys.contains(metric), String.format("Metric '%s' cannot be used as an output metric as no plugin declare this metric", metric));
+      return true;
+    }
+  }
+
+  private class ValidateUniqueOutputMetric implements Predicate<MeasureComputerWrapper> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)